firebase-ai/src/main/kotlin/com/google/firebase/ai/type/ThinkingConfig.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.ai.type
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+/** Configuration parameters for thinking features. */
+public class ThinkingConfig(public val thinkingBudget: Int?) {
+
+  public class Builder() {
```

Comment:
Since it's a single parameter now, could we do away with the builder and add it later when it becomes necessary?

Reply:
My concern is that if we later need to re-order the parameters for some reason it would be a source incompatible change for Java (and Kotlin without named arguments).
I would rather incentivize Java developers to use the builder from the get-go.

Reply:
SGTM as nullable